.github/workflows/pythonpackage.yml

```diff
@@ -12,16 +12,20 @@ jobs:
       max-parallel: 4
       matrix:
         python-version: [3.6, 3.7]
-        os: [ubuntu-16.04, windows-latest, macos-latest]
+        os: [ubuntu-18.04, windows-latest, macos-latest]
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
+      with:
+          # Fetch all history so that setuptool_scm can build the correct version string.
+          fetch-depth: 0
+
     - name: setup-conda
       uses: s-weigand/setup-conda@v1.0.0
```

Comment:
This is a great question and I have some updates about this. But I'm leaning towards Rich's comments that maybe that should be a separate issue and PR, because it's currently complicated. Complicated in that `setup-miniconda` works for `mac` and `linux` but now fails on `windows` because of how Miniconda is added to Windows Path.

Reply:
Uffda.  Sounds great to have this be in a separate PR!